### PR TITLE
fix(FEC-10687): Allow partial config in setMedia API

### PR DESCRIPTION
**Pull request by dan-ziv**
### Description of the Changes

When using `setMedia` API, if plugins object isn't set the player is crashing.
Create the plugins object internal if isn't exists.

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/kaltura-player.js [hunk 1]
```diff
@@ -131,6 +131,7 @@ class KalturaPlayer extends FakeEventTarget {
     const playerConfig = Utils.Object.copyDeep(mediaConfig);
     Utils.Object.mergeDeep(playerConfig.sources, this._localPlayer.config.sources);
     Utils.Object.mergeDeep(playerConfig.session, this._localPlayer.config.session);
+    playerConfig.plugins = playerConfig.plugins || {};
```
Comment (OrenMe) [inline src/kaltura-player.js:134]: is this reflected in the `KPMediaConfig` object?

**Review by OrenMe — APPROVED** [review]

File: src/kaltura-player.js (hunk 1 above)
Comment (dan-ziv) [reply src/kaltura-player.js:138]: should be change in provider types

File: src/common/utils/thumbs.js [hunk 2]
```diff
@@ -10,18 +10,18 @@ const TEMPLATE: string = '{{thumbnailUrl}}/width/{{width}}/vid_slices/{{slices}}
...
-export function getThumbSlicesUrl(mediaConfig: ProviderMediaConfigObject, seekbarConfig?: SeekbarConfig): string {
+export function getThumbSlicesUrl(mediaConfig: KPMediaConfig, seekbarConfig?: SeekbarConfig): string {
   const mediaConfigPoster = mediaConfig.sources && mediaConfig.sources.poster;
   if (typeof mediaConfigPoster === 'string') {
     if (THUMBNAIL_REGEX.test(mediaConfigPoster)) {
       try {
         const model: Object = {
           thumbnailUrl: mediaConfigPoster,
-          ks: mediaConfig.session.ks,
+          ks: mediaConfig.session && mediaConfig.session.ks,
```
Comment (OrenMe) [inline src/common/utils/thumbs.js:24]: WDYS about adding [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)?
```suggestion
          ks: mediaConfig.session?.ks,
```

Comment (dan-ziv) [reply src/common/utils/thumbs.js:24]: nice

File: src/common/utils/thumbs.js [hunk 3]
```diff
@@ -10,18 +10,18 @@ const TEMPLATE: string = '{{thumbnailUrl}}/width/{{width}}/vid_slices/{{slices}}
...
-export function getThumbSlicesUrl(mediaConfig: ProviderMediaConfigObject, seekbarConfig?: SeekbarConfig): string {
+export function getThumbSlicesUrl(mediaConfig: KPMediaConfig, seekbarConfig?: SeekbarConfig): string {
   const mediaConfigPoster = mediaConfig.sources && mediaConfig.sources.poster;
   if (typeof mediaConfigPoster === 'string') {
     if (THUMBNAIL_REGEX.test(mediaConfigPoster)) {
       try {
         const model: Object = {
           thumbnailUrl: mediaConfigPoster,
-          ks: mediaConfig.session.ks,
+          ks: mediaConfig.session?.ks,
```
Comment (OrenMe) [inline src/common/utils/thumbs.js:24]: We probably need a Babel plugin for this to be transpiled. Let's check that.

Comment (dan-ziv) [reply src/common/utils/thumbs.js]: Not critical for now, I will keep my original code.

**Review by OrenMe — APPROVED** [review]

**Pull request merged**